pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -1538,6 +1538,21 @@ func (r *ReconcilePerconaServerMongoDB) reconcileStatefulSet(
 		sfs.Labels = customLabels
 	}
 
+	pods, err := psmdb.GetRSPods(ctx, r.client, cr, replset.Name, true)
+	if err != nil {
+		return nil, errors.Wrap(err, "get rs pods")
+	}
+
+	if len(pods.Items) > int(replset.Size+replset.Arbiter.Size+replset.NonVoting.Size) {
```

Comment:
what if arbiter or nonvoting is disabled (`enabled: false`) but their size 3?

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1116/commits/7d6001bbed7906a75eb6cb4207bbf881c57d0fd9